prime-dotnet-webapi/Services/EnrolleeService.cs

```diff
@@ -495,6 +497,19 @@ private async Task CreateSelfDeclarationDocuments(int enrolleeId, ICollection<Se
                     });
                 }
             }
+
+            foreach (var document in currentSelfDeclarationDocuments)
+            {
+                if (!newDeclarations.Select(nd => nd.SelfDeclarationTypeCode).Contains(document.SelfDeclarationTypeCode))
+                {
+                    var documentToUpdate = await _context.SelfDeclarationDocuments
+                        .Where(sdd => sdd.EnrolleeId == enrolleeId && sdd.DocumentGuid == document.DocumentGuid)
+                        .SingleOrDefaultAsync();
```

Comment:
you don't need this call, right? you already have the document

Reply:
`currentSelfDeclarationDocuments` is a list of `SelfDeclarationDocumentViewModel`. When I tried making changes to it those changes weren't reflected in the model itself. I may be missing something.

Reply:
I didn't realize that returned View Models. Yes, if you update a view model it wont change the model/entity in the DB. But then why get the view models and then each model individually when you could just fetch the models once?

Reply:
Also, if you did want to do it this way, why retrieve the models by doing `.Where(sdd => sdd.EnrolleeId == enrolleeId && sdd.DocumentGuid == currentDocument.DocumentGuid)` instead of just `.Where(sdd => sdd.Id == currentDocument.Id)`?

Reply:
The reason I got the view models was because I wanted to reuse the existing `GetSelfDeclarationDocumentsAsync` method and then just flag the documents that we want to hide, We could write another method that gets the models, change the current method to return models, or just do the query in place. 

As to the second question, I guess that was a much better solution that I derped out on lol.

Reply:
I am leaning towards changing the current `GetSelfDeclarationDocumentsAsync ` method, and instead of projecting the result inside it to `SelfDeclarationDocumentViewModel` we can instead do the projection where the view model is needed, for now that would be in the controller.

Reply:
@james-hollinger please take a look at the changes. I used automapper in the controller and removed the projection from the service method.